src/kademlia/traversal_algorithm.cpp

```diff
@@ -593,6 +593,24 @@ void traversal_algorithm::status(dht_lookup& l)
 	l.last_sent = last_sent;
 }
 
+void look_for_nodes(node const& node, bdecode_node const& r, std::function<void(const node_endpoint&)> f)
```

Comment:
I really think the "nodes_key" should be the parameter here, not the whole node. Imagine writing a unit test for this function. As it stands now you would have to construct a whole DHT node object, just in order for this function to be able to query for the ``protocol_nodes_key()`` from it. It makes the function have a lot more dependencies than it needs.

If you instead just pass in a ``char const*`` or ``string_view`` for the actual key name, it would be a lot simpler to test.